connector/src/main/java/org/jboss/as/connector/subsystems/datasources/JdbcDriverAdd.java

```diff
@@ -97,14 +97,6 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final ModuleIdentifier moduleId;
         final Module module;
         String slot = operation.hasDefined(MODULE_SLOT.getName()) ? MODULE_SLOT.resolveModelAttribute(context, operation).asString() : null;
-        if (moduleName.contains(":")) {
-            slot = moduleName.substring(moduleName.indexOf(":") + 1);
-            moduleName = moduleName.substring(0, moduleName.indexOf(":"));
-        } else {
-            if (slot != null) {
-                model.get(DRIVER_MODULE_NAME.getName()).set(moduleName + ":" + slot);
-            }
-        }
```

Comment:
moduleName here should not contain ":" in any case, and we don't need to change the "driver-module-name" of the  ModelNode, since we have the "module_slot" defined already.